packages/vue-renderer/src/renderer.js

```diff
@@ -438,15 +438,17 @@ export default class VueRenderer {
     // Add url to the context
     context.url = url
 
+    const { req = {} } = context
+
     // context.spa
     if (context.spa === undefined) {
       // TODO: Remove reading from context.res in Nuxt3
-      context.spa = !this.SSR || context.spa || (context.req && context.req.spa) || (context.res && context.res.spa)
+      context.spa = !this.SSR || context.spa || req.spa || (context.res && context.res.spa)
     }
 
-    // context.modern
-    if (context.modern === undefined) {
-      context.modern = context.req ? (context.req.modernMode || context.req.modern) : false
+    // context.useModernRenderer
```

Comment:
I think `context.modern` is more consistent with other context keys. Also allows passing an object instead of Boolean for behavior changing in future

Reply:
It makes sense, I was just thinking it may be kind of confusing with options.modern.

Reply:
Yes. That's also a note. But maybe this change worth for consistency.